src/meson.build

```diff
@@ -44,6 +44,62 @@ extendedWarnings = [
 	'-Wstack-protector',
 	'-Wunsuffixed-float-constant',
 	'-Wimplicit-fallthrough',
+	'-Wundef',
+	'-Wignored-qualifiers',
+	'-Wshadow',
+	'-Wswitch-enum',
+	'-Wdefaulted-function-deleted',
+	'-Wdeprecated-copy',
+
+	# MSVC unique warnings that are useful
+	'-w35030', # attribute 'gnu::packed' is not recognized
```

Comment:
Do we want this enabled or disabled? We'd have thought the latter because the attribute does nothing if the compiler doesn't support it and it comes from the `gnu` compiler-specific attribute namespace.
Perhaps if we can find a `msvc` namespace equivalent we can apply that to get consistent behaviour across compilers?

Reply:
No, this should be detected with `if __has_attribute(gnu::packed) || defined(__GNU__)` and the relevant `define` applied.

Reply:
Having talked about this with you in DMs, agreed. Specifically because MSVC doesn't support a method other than a compiler-specific pragma, and the attribute being required to generate correct code thanks to having to deal with raw USB-specified structure layout that ignored packing requirements for the types used